bottle.py

```diff
@@ -579,9 +579,10 @@ def get_undecorated_callback(self):
         func = self.callback
         func = getattr(func, '__func__' if py3k else 'im_func', func)
         closure_attr = '__closure__' if py3k else 'func_closure'
-        while hasattr(func, closure_attr) and getattr(func, closure_attr):
-            attributes = getattr(func, closure_attr)
+        attributes = getattr(func, closure_attr, False)
+        while attributes:
```

Comment:
The original code does not define `attributes` if ` hasattr(func, closure_attr)` isn't true.
Your change always defines it.
I suspect (have not benchmarked it) that ` hasattr(func, closure_attr)` is faster than the creation of `attribute = Flase`. Hence, I would rather keep the original code untouched.

Reply:
get_undecorated_callback() serves to one goal - to be called for repr(callback), ie for debug via `print()`. Nothing critical for speed.